### DISCUSSION TITLE
I want to move node contains child nodes

### DISCUSSION THREAD

**Discussion by hayamat** [fremail/sequelize-nested-set#25] [Q&A]
I have a question about the library sequelize-nested-set
I want to move node contains child nodes
Currently, there is only a way to get the each id of the node to move to the target one by one. 

sample image
![image](https://user-images.githubusercontent.com/6092384/126683615-a4e974cc-a13e-4fd8-a66f-4733d2b1ef74.png)

**Comment by AntonDemushkin** [comment]
Hi @hayamat, thank you for the question.

All methods to move a record moves the record and all its descendants. The methods are listed here: https://github.com/fremail/sequelize-nested-set/wiki/Moving-a-record

Your Japanese example is too hard for me, sorry. Let's use a simplified one:
-- 1
--- 1.1
---- 1.1.1
--- 1.2

You want to move 1.1 (and 1.1.1) to be after 1.2, right? So the tree should look this way after update?

-- 1
--- 1.2
--- 1.1
---- 1.1.1


Then you can use `moveAsNextSiblingOf` or `moveAsLastChildOf`.

```js
await one_one.moveAsNextSiblingOf(one_two);
// or
await one_one.moveAsLastChildOf(one);
```

**Reply by hayamat** [reply]
Hi, thank you for your reply.
The code I received didn't work.

Is there something wrong?

tree image
![image](https://user-images.githubusercontent.com/6092384/126885506-b066b6f0-30f3-40af-9dc8-12d74f3c0c0c.png)

Data
![image](https://user-images.githubusercontent.com/6092384/126885518-cac62226-4051-4a7a-899d-96cfd98df99a.png)

soruce code
![image](https://user-images.githubusercontent.com/6092384/126885523-75a9b9e0-410a-4ae9-a4f7-2642ed1e83b8.png)

console error
![image](https://user-images.githubusercontent.com/6092384/126885531-330fbc9b-86f2-41d9-af30-9d1eb18fb606.png)

**Reply by AntonDemushkin** [reply]
Oh! @hayamat, I got your problem. You have incorrect values of `lft` and `rgt` fields.

You can check the theory of nested set model in Wikipedia: https://en.wikipedia.org/wiki/Nested_set_model#Example

The correct values for your example must be so:
![image](https://user-images.githubusercontent.com/12110440/126897713-2efc0f6a-a42c-47fc-b40b-28da52e7b003.png)

**Reply by hayashitvb** [reply]
The test data was wrong.
Thank you for your advice!

**Reply by hayamat** [reply]
Thank You 😭
